### PR TITLE
Feature : concept / routing

### PR DESCRIPTION
**Pull request by maissani**
# Before validation
- [X] Default route
- [X] Before Accessing Controller
- [ ] SLUG routes ?

Need to be discuss and approbate by maintainers.

### CODE REVIEW

File: concepts/routing/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// config/routes.js
+module.exports = [
+  {
+    method: [ 'GET' ],
+    path: '/example/test',
+    handler: 'ExampleController.test'
+  }
+]
+```
+> method: can be ['POST', 'GET', 'PUT', 'DELETE']
```
Comment (jaumard) [inline concepts/routing/README.md]: There (at least) also patch, head, option

File: concepts/routing/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+> handler: The controller that handle the request, in most of case: ControllerClass.ControllerFunction
+
+##### Execute Something before accessing Controller
+```JavaScript
+{
+  method: [ 'GET' ],
+  path: '/example/test',
+  handler: 'ExampleController.test',
+  config: {
+    pre: [ 'ExamplePreClass.test' ]
```
Comment (jaumard) [inline concepts/routing/README.md]: We don't want users to config policies this way, there a `config.policies` for this

**Comment by jaumard** [comment]
Also there some additional configuration that can be use look here https://github.com/trailsjs/trailpack-router/blob/master/lib/schemas/route.js all is working with hapi for now express doesn't support other fields (validate is under development)

**Comment by maissani** [comment]
ok, next update in a couple of hours..

File: concepts/routing/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    method: [ 'GET' ],
+    path: '/example/test',
+    handler: 'ExampleController.test'
+  }
+]
+```
+
+###### Required values
+
+> method: can be ['POST', 'GET', 'PUT', 'DELETE']
```
Comment (jaumard) [inline concepts/routing/README.md]: not only this ^^ OPTIONS, PATCH, HEAD are available too (don't remember if there others)

**Comment by maissani** [comment]
Ok, i ll update it this afternoon.

**Comment by maissani** [comment]
Should i had theses methods ? 

### Express Methods :
- checkout
- copy
- delete
- get
- head
- lock
- merge
- mkactivity
- mkcol
- move
- m-search
- notify
- options
- patch
- post
- purge
- put
- report
- search
- subscribe
- trace
- unlock
- unsubscribe

### Hapi

You can provide whatever you want in method

### Koa

You can provide whatever you want in method

**Comment by jaumard** [comment]
From express docs here http://expressjs.com/fr/guide/routing.html#route-methods 

> Express prend en charge les méthodes de routage suivantes qui correspondent aux méthodes HTTP : get, post, put, head, delete, options, trace, copy, lock, mkcol, move, purge, propfind, proppatch, unlock, report, mkactivity, checkout, merge, m-search, notify, subscribe, unsubscribe, patch, search, and connect.

**Comment by maissani** [comment]
So it contains more methods for express than i said (Strangely because i have see the same page of the doc :p ): 

I see the docs for express 
- get
- post
- put
- head
- delete
- options
- trace
- copy
- lock
- mkcol
- move
- purge
- propfind
- proppatch
- unlock
- report
- mkactivity
- checkout
- merge
- m-search
- notify
- subscribe
- unsubscribe
- patch
- search
- connect

**Comment by maissani** [comment]
I update now and then you will be able to merge it

**Comment by jaumard** [comment]
In fact I just think that if we want docs part of trailpacks, this docs should only explain what is a route and what it does in Trails, also explain the validations schema in details and say `this validations schema can be partially supported by the webserver so please check your web server documentations [links....]`

What do you think ? Each native web server doc is great but should be move under each trailpack

**Comment by maissani** [comment]
@jaumard i ll close this PR and not merge it cause of docs is moving respectively under respective trailpack, this repo will forward to each part of docs

**Comment by jaumard** [comment]
@maissani but trails-docs should talk about general feature and dispatch to correct trailpack docs so you can just modify it and then we can merge it

**Comment by connor11528** [comment]
lgtm

**Comment by jaumard** [comment]
@maissani routes should move into specific trailpack-hapi/express and here we should have links only :)

**Pull request closed without merging**